app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -18,6 +18,56 @@ $(function () {
     $('#ldap_conn_message').html(msg)
   }
 
+  function ldapConnectionTestPart1 () {
+    var file = $('#dex_connector_ldap_certificate').val()
+    if(!file) {
+      ldapConnectionTestPart2( $('#dex_connector_ldap_current_cert').val() )
```

Comment:
Note that despite this code getting the current cert contents and using them for testing, "saving" is still broken if a new cert is not chosen via the file upload control. This is a bad user experience and is bound to cause confusion. A bug will be filed for that and a different PR made to address that issue.

Reply:
If the user doesn't select a file, use the current one for testing the connection. If there's no current cert then the test would fail as expected, no? That's right, no? 

What do you mean by "still broken"? Do you mean that there's no way of saving if the test connection fail?

Reply:
To clarify, the following sequence is a problem:
1. Create an ldap connector ( enter all data, test connection, save )
2. Go to edit the connector ( cert is shown )
3. Change something ( perhaps reconfigure the port )
4. Re-test the connection ( would work; pulling cert from the displayed textarea )
5. Try to save ( you can't; errors out )

That is, this PR allows the 'test connection' button to work in the edit scenario, but the edit scenario still cannot be saved through currently. This PR does not fix that problem.

--

Besides that there is another existing bug with the current code:
1. Enter correct details into an ldap connector
2. Click 'test connection' and succeed; save is enabled
3. Alter the details to be incorrect ( save is not properly then re-disabled )
4. Save the bad data

This scenario also is not corrected by this PR.
Both of the scenarios I am describing are separate bugs that need to be resolved.
This PR is only meant to address the fact that even the 'happy path' for ldap external authentication is currently broken in HEAD.

Reply:
The latest updates to this PR fix the first scenario. Saving of an existing connector after changes without re-uploading the cert now works.